### ISSUE TITLE
Allow inherit from `BaseButton` when creating new script

### ISSUE THREAD

**Issue by acgc99** [godotengine/godot-proposals#8101]
### Describe the project you are working on

A plugin for app development [widgets](https://github.com/acgc99/Godot-Widgets).

### Describe the problem or limitation you are having in your project

Multiple custom nodes inherit from `BaseButton`, however you cannot create a script that  inherits from `BaseButton` from the `FileSystem` tab.

![image](https://github.com/godotengine/godot-proposals/assets/96864661/5c3170d1-b80f-4d11-acf3-84a1b1d8089b)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add the option to inherit from `BaseButton`.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add the option to inherit from `BaseButton`.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes, you can modify the created script.

### Is there a reason why this should be core and not an add-on in the asset library?

Yes, I think it is a limitation saying "you shouldn't do that".

**Comment by AThousandShips** [comment]
Base button doesn't contain much of the logic required

You are not providing any reasons for being able to extend this, please fill in the form properly, all entries must be filled in:
> Describe the feature / enhancement and how it helps to overcome the problem or limitation

This one should be filled in with what you need, not what to do

**Comment by Stronkkey** [comment]
@AThousandShips I think they mean that you cannot pick abstract classes to inherit from in the script creation wizard.
![abstract](https://github.com/godotengine/godot-proposals/assets/92468390/72edd306-adfc-43bf-8656-841fad178459)

**Comment by AThousandShips** [comment]
Yes, that's by design, you're not supposed to be able to extend these classes, that's not a bug, `BaseButton` is not a complete class and doesn't work on its own

**Comment by acgc99** [comment]
Therefore, things like `extends BaseButton` should be forbidden.

**Comment by AThousandShips** [comment]
See the discussion in https://github.com/godotengine/godot-proposals/issues/5641

**Comment by acgc99** [comment]
Is there any reason why I shouldn't use it specifically? For example, this is `WIconButton` structure (a button that is represented by an texture (only one texture, not like `TextureButton`)):
- BaseButton
  - TextureRect

And it works as I expect. I think that is better than:
- Control
  - Button

which I would use to avoid exposing properties like `Button.text` on the inspector if `BaseButton` is forbidden.

**Comment by AThousandShips** [comment]
As far as I recall generally the `BaseButton` class is meant as an abstract base which doesn't handle the main behaviour, but I might be wrong, but generally the current implementation doesn't handle abstract classes completely so there's limitations to how it works, that's why it's hidden

I asked from the start why you needed this, and that this information should have been added to the proposal as that's what the instructions say, that's exactly why I asked...

**Comment by acgc99** [comment]
@AThousandShips sorry... I editted the issue.

**Comment by gregcsokas** [comment]
But with a script, you can create any kind of button.

Like here
https://github.com/gregcsokas/godot-ui-components/blob/main/templates/text_button/text_button.gd

Unfortunately, it's laggy as hell

**Comment by kleonc** [comment]
This looks like a bug to me. Since https://github.com/godotengine/godot/pull/58972 there's a distinction in binding between "abstract" and "virtual" classes. Quote:

> * Added a new "virtual" category for classes, **they can't be instantiated from the editor, but can be inherited from script** and extensions (GDREGISTER_VIRTUAL_CLASS).

If "virtual" classes (like BaseButton) are said to be extendable from script then I see no reason to disallow picking them as a base class when creating a new script from the FileSystem dock.

In case of adding to the scene tree a scriptless instance of a "virtual" class within the Scene dock I'm not sure why would it need to be forbidden. But even if it's forbidden for some sensible reason note that it's easily workaroundable, example:

```gdscript
class_name Test
extends BaseButton
```
1. Empty scene -> Create root node -> Other node -> `Test`.
![PGkua3jrDj](https://github.com/godotengine/godot-proposals/assets/9283098/1639e793-6b1b-4602-82aa-b46d82656d69)

2. Detach script, you have BaseButton scene with no script.
![g1e19zlR1i](https://github.com/godotengine/godot-proposals/assets/9283098/4b60bb84-c248-40af-aec1-726fc2de75ad)

3. Run the scene. No explosion. :thinking:

Not sure if e.g. detaching a script from a "virtual" class should be forbidden or what. For sure there are many inconsistencies in the logic of handling such classes.
